### PR TITLE
Adding scenario: web app integrated with app gateway using service endpoint

### PR DESCRIPTION
**Pull request by madsd**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Initial creation

### CODE REVIEW

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+            "defaultValue": "myAppGatewaySubnet",
+            "type": "string"
+        },
+        "appGatewayName": {
+            "defaultValue": "myAppGateway",
+            "type": "string"
+        }
+     },
+    "variables": {  
+        "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        "virtualNetworkName": "[concat(parameters('virtualNetworkName'), '-', uniqueString(resourceGroup().id))]",
+        "virtualNetworkAddressPrefix": "10.0.0.0/20",
+        "virtualNetworkSubnetName": "[parameters('appGatewaySubnetName')]",
+        "virtualNetworkSubnetPrefix": "10.0.0.0/24",
+        "virtualNetworkId": "[resourceId(resourceGroup().name, 'Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]",
+        "virtualNetworkSubnetId": "[concat(variables('virtualNetworkId'), '/subnets/', variables('virtualNetworkSubnetName'))]",
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: remove concat and use resourceId

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        "applicationGatewaySkuSize": "Standard_v2",
+        "applicationGatewayTier": "Standard_v2",
+        "applicationGatewayAutoScaleMinCapacity": 2,
+        "applicationGatewayAutoScaleMaxCapacity": 5,
+        "appGwIpConfigName": "appGatewayIpConfigName",
+        "appGwFrontendPortName": "appGatewayFrontendPort_80",
+        "appGwFrontendPort": 80,
+        "appGwFrontendPortId": "[concat(variables('applicationGatewayId'), '/frontendPorts/', variables('appGwFrontendPortName'))]",
+        "appGwFrontendIpConfigName": "appGatewayPublicFrontendIpConfig",
+        "appGwFrontendIpConfigId": "[concat(variables('applicationGatewayId'), '/frontendIPConfigurations/', variables('appGwFrontendIpConfigName'))]",
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: remove concat and use resourceId

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+        "applicationGatewayName": "[concat(parameters('appGatewayName'), '-', uniqueString(resourceGroup().id))]",
+        "applicationGatewayId": "[resourceId('Microsoft.Network/applicationGateways', variables('applicationGatewayName'))]",
+        "applicationGatewaySkuSize": "Standard_v2",
+        "applicationGatewayTier": "Standard_v2",
+        "applicationGatewayAutoScaleMinCapacity": 2,
+        "applicationGatewayAutoScaleMaxCapacity": 5,
+        "appGwIpConfigName": "appGatewayIpConfigName",
+        "appGwFrontendPortName": "appGatewayFrontendPort_80",
+        "appGwFrontendPort": 80,
+        "appGwFrontendPortId": "[concat(variables('applicationGatewayId'), '/frontendPorts/', variables('appGwFrontendPortName'))]",
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: remove concat and use resourceId

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+        "appGwFrontendIpConfigName": "appGatewayPublicFrontendIpConfig",
+        "appGwFrontendIpConfigId": "[concat(variables('applicationGatewayId'), '/frontendIPConfigurations/', variables('appGwFrontendIpConfigName'))]",
+        "appGwHttpSettingName": "appGatewayHttpSetting_80",
+        "appGwHttpSettingId": "[concat(variables('applicationGatewayId'), '/backendHttpSettingsCollection/', variables('appGwHttpSettingName'))]",
+        "appGwHttpSettingProbeName": "appGatewayHttpSettingProbe_80",
+        "appGwBackendAddressPoolName": "[concat('appGateway', variables('webAppName'), 'BackendPool')]",
+        "appGwBackendAddressPoolId": "[concat(variables('applicationGatewayId'), '/backendAddressPools/', variables('appGwBackendAddressPoolName'))]",
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: remove concat and use resourceId

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+        "appGwListenerName": "appGatewayListener",
+        "appGwListenerId": "[concat(variables('applicationGatewayId'), '/httpListeners/', variables('appGwListenerName'))]",
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: remove concat and use resourceId

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+                    "priority": 200,
+                    "name": "appGatewaySubnet",
+                    "description": "Isolate traffic to subnet containing Azure Application Gateway"
+                    }
+                ]
+              },
+              "httpsOnly": "false"
+            },
+            "identity": {
+                "type": "SystemAssigned"
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: how is this used?

File: 201-web-app-with-app-gateway-v2/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+        }
+    ],
+    "outputs": {
+        "appGatewayUrl": {
+            "type": "string",
+            "value": "[concat('http://', reference(variables('publicIpAddressName')).dnsSettings.fqdn, '/')]"
+        },
+        "webAppUrl": {
+            "type": "string",
+            "value": "[concat('http://', toLower(variables('webAppName')), '.azurewebsites.net/')]"
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.json]: use reference() to get the url

File: 201-web-app-with-app-gateway-v2/azuredeploy.parameters.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": { 
+        "webAppName": {
```
Comment (bmoore-msft) [inline 201-web-app-with-app-gateway-v2/azuredeploy.parameters.json]: remove these since you have defaultValues

Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.parameters.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 8 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 7 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: It is not, but were just in the interest of being prepared. I have removed it.

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 6 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 5 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 4 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 3 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 2 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

File: 201-web-app-with-app-gateway-v2/azuredeploy.json (hunk 1 above)
Comment (madsd) [reply 201-web-app-with-app-gateway-v2/azuredeploy.json]: done

**Comment by madsd** [comment]
Hi @bmoore-msft . Thank you for your review and comments. Great feedback! I implemented it and everything is now green :-) Let me know if you have more input.
Br, Mads

**Comment by madsd** [comment]
Hi @bmoore-msft . Anything else I need to do for this request to get merged?

**Pull request merged**